File: decompiled/sources/freemarker/template/ObjectWrapper.java
package freemarker.template;

import freemarker.ext.beans.BeansWrapper;

public interface ObjectWrapper {
    public static final ObjectWrapper BEANS_WRAPPER = BeansWrapper.getDefaultInstance();
    public static final ObjectWrapper DEFAULT_WRAPPER = DefaultObjectWrapper.instance;
    public static final ObjectWrapper SIMPLE_WRAPPER = SimpleObjectWrapper.instance;

    TemplateModel wrap(Object obj) throws TemplateModelException;
}

File: decompiled/sources/freemarker/template/ObjectWrapper.kt
package freemarker.template

import freemarker.ext.beans.BeansWrapper

interface ObjectWrapper {
    @Throws(TemplateModelException::class)
    fun wrap(obj: Any?): TemplateModel?

    companion object {
        @JvmField
        val BEANS_WRAPPER: ObjectWrapper = BeansWrapper.getDefaultInstance()

        @JvmField
        val DEFAULT_WRAPPER: ObjectWrapper = DefaultObjectWrapper.instance

        @JvmField
        val SIMPLE_WRAPPER: ObjectWrapper = SimpleObjectWrapper.instance
    }
}
